pkgs/development/compilers/graalvm/default.nix

```diff
@@ -1,31 +1,86 @@
-{ stdenv, lib, fetchFromGitHub, fetchurl, fetchzip, mercurial, python27, zlib, makeWrapper, oraclejdk8 }:
+{ stdenv, lib, fetchFromGitHub, fetchurl, fetchzip, fetchgit, mercurial, python27,
+  zlib, makeWrapper, openjdk, unzip, git, clang, llvm, which, icu }:
 
 let
+  truffleMake = ./truffle.make;
+  makeMxGitCache = list: out: ''
+     mkdir ${out}
+    ${lib.concatMapStrings ({ url, name, rev, sha256 }: ''
+      mkdir -p ${out}/${name}
+      cp -rf ${fetchgit { preferLocalBuild = false;
+                          deepClone = true;
+                          leaveDotGit = true;
```

Comment:
```leaveDotGit = true;``` is not fully deterministic, the ```sha256``` on another machine might vary.
that's why here I created fake ```hg``` repos to fool ```mx```

Reply:
Ah good catch, I had it there before I totally patched the revision check mechanism from mx out, I'll remove this, shouldn't affect anything.

Reply:
Fixed